lib/std/http.zig

```diff
@@ -991,44 +961,32 @@ pub const BodyWriter = struct {
             return error.Unimplemented;
         };
         const out = bw.http_protocol_output;
-        const chunked = &bw.state.chunked;
-        state: switch (chunked.*) {
-            .offset => |off| {
-                // TODO: is it better perf to read small files into the buffer?
-                const buffered_len = out.end - off - chunk_header_template.len;
-                const chunk_len = data_len + buffered_len;
-                writeHex(out.buffer[off..][0..chunk_len_digits], chunk_len);
+        switch (bw.state.chunk_len) {
+            0 => {
+                const header_buf = try out.writableArray(chunk_header_template.len);
+                @memcpy(header_buf, chunk_header_template);
+                writeHex(header_buf[0..chunk_len_digits], data_len);
                 const n = try out.sendFileHeader(w.buffered(), file_reader, limit);
-                chunked.* = .{ .chunk_len = data_len + 2 - n };
-                return w.consume(n);
+                bw.state.chunk_len = data_len + 2 - n;
+                const ret = w.consume(n);
+                return ret;
             },
-            .chunk_len => |chunk_len| l: switch (chunk_len) {
-                0 => {
-                    const off = out.end;
-                    const header_buf = try out.writableArray(chunk_header_template.len);
-                    @memcpy(header_buf, chunk_header_template);
-                    chunked.* = .{ .offset = off };
-                    continue :state .{ .offset = off };
-                },
-                1 => {
-                    try out.writeByte('\n');
-                    chunked.chunk_len = 0;
-                    continue :l 0;
-                },
-                2 => {
-                    try out.writeByte('\r');
-                    chunked.chunk_len = 1;
-                    continue :l 1;
-                },
-                else => {
-                    const chunk_limit: std.Io.Limit = .limited(chunk_len - 2);
-                    const n = if (chunk_limit.subtract(w.buffered().len)) |sendfile_limit|
-                        try out.sendFileHeader(w.buffered(), file_reader, sendfile_limit.min(limit))
-                    else
-                        try out.write(chunk_limit.slice(w.buffered()));
-                    chunked.chunk_len = chunk_len - n;
-                    return w.consume(n);
-                },
+            1 => unreachable,
```

Comment:
is this really unreachable? it looks like it could be reached from line 970 setting chunk_len to 1, or line 987 setting chunk_len to 1.

Reply:
Yes it is unreachable. It is in fact an assertion that we do not violate the protocol by e.g. only `\n` rather than `\r\n`. I do see a way to make this code a bit more clear though.